webui/react/src/pages/ExperimentDetails/TrialCheckpoints.tsx

```diff
@@ -0,0 +1,418 @@
+            value,
+          }));
+      } else if (column.key === Checkpointv1SortBy.SEARCHERMETRIC) {
+        column.render = HumanReadableNumberRenderer;
+        column.title = `Searcher Metric (${experiment.config.searcher.metric})`;
+      } else if (column.key === 'actions') {
+        column.render = actionRenderer;
+      }
+      if (column.key === settings.sortKey) {
+        column.sortOrder = settings.sortDesc ? 'descend' : 'ascend';
+      }
+      return column;
+    });
+
+    return newColumns;
+  }, [
+    dropDownOnTrigger,
+    experiment.config.searcher.metric,
+    handleOpenCheckpoint,
+    settings.sortDesc,
+    settings.sortKey,
+    stateFilterDropdown,
+  ]);
+
+  const fetchTrialCheckpoints = useCallback(async () => {
+    if (!settings) return;
+
+    const states = settings.state?.map((state) => encodeCheckpointState(state as CheckpointState));
+    try {
+      const response = await getTrialCheckpoints(
+        {
+          id: trial.id,
+          limit: settings.tableLimit,
+          offset: settings.tableOffset,
+          orderBy: settings.sortDesc ? 'ORDER_BY_DESC' : 'ORDER_BY_ASC',
+          sortBy: validateDetApiEnum(Checkpointv1SortBy, settings.sortKey),
+          states: validateDetApiEnumList(Checkpointv1State, states),
+        },
+        { signal: canceler.signal },
+      );
+      setTotal(response.pagination.total ?? 0);
+      if (!isEqual(response.checkpoints, checkpoints)) {
+        setCheckpoints(response.checkpoints);
+      }
+    } catch (e) {
+      handleError(e, {
+        publicSubject: `Unable to fetch ${f_flat_runs ? 'run' : 'trial'} ${trial.id} checkpoints.`,
+        silent: true,
+        type: ErrorType.Api,
+      });
+    } finally {
+      setIsLoading(false);
+    }
+  }, [f_flat_runs, settings, trial.id, canceler.signal, checkpoints]);
+
+  const submitBatchAction = useCallback(
+    async (action: CheckpointAction) => {
+      if (!settings.row) return;
+      try {
+        dropDownOnTrigger(settings.row)[action]();
+
+        // Refetch experiment list to get updates based on batch action.
+        await fetchTrialCheckpoints();
+      } catch (e) {
+        const publicSubject = `Unable to ${action} Selected Checkpoints`;
+        handleError(e, {
+          level: ErrorLevel.Error,
+          publicMessage: 'Please try again later.',
+          publicSubject,
+          silent: false,
+          type: ErrorType.Server,
+        });
+      }
+    },
+    [dropDownOnTrigger, fetchTrialCheckpoints, settings.row],
+  );
+
+  const { stopPolling } = usePolling(fetchTrialCheckpoints, { rerunOnNewFn: true });
+
+  // Get new trials based on changes to the pagination, sorter and filters.
+  useEffect(() => {
+    setIsLoading(true);
+    fetchTrialCheckpoints();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    return () => {
+      canceler.abort();
+      stopPolling();
+    };
+  }, [canceler, stopPolling]);
```

Comment:
`usePolling` handles calling the polling function and stopping on unmount
```suggestion
  usePolling(fetchTrialCheckpoints, { rerunOnNewFn: true });
  
  useEffect(() => {
    return () => {
      canceler.abort();
    };
  }, [canceler]);
```